service/service.go

```diff
@@ -234,8 +234,23 @@ func (svc *Service) Start(wg *sync.WaitGroup, c chan os.Signal) {
 						log.Info("Payment Details Response : ",
 							log.Data{keys.PaymentDetails: paymentDetails, keys.StatusCode: statusCode})
 
-						//Filter accepted payments from GovPay
-						if paymentDetails.PaymentStatus == "accepted" {
+						if isRefundTransaction(pp) {
+							refund := data.RefundResource{}
+							for _, ref := range paymentResponse.Refunds {
+								if ref.RefundId == pp.RefundId {
+									refund = ref
+								}
+							}
```

Comment:
1. this can be improved from a performance point of view if moved into a private function:
```
func getRefund()... {
  for _, ref := range paymentResponse.Refunds {
    if ref.RefundId == pp.RefundId {
      return ref
    }
  }

  // error if refund cant be found?
}
```
this way as soon a match is found, it exits the loop rather than carrying on with the remaining iterations.

2. What if no refund is found? Your code does not accommodate this anywhere. Please handle this and add a unit test.

3. I think we've added too many lines to an already huge function. I think we should minimise extensions to this function with something like:
```
if isRefundTransaction(pp) {
  handleRefundTransaction(...)
} else if paymentDetails.PaymentStatus == "accepted" {
  ...
}
```

Reply:
2. If no refund is found, the status will not be "success" and we will not go into the if block.

Reply:
But will `null.Status` trigger a Null Pointer or whatever GoLang has instead?